docs/admin/kubeadm-init.md

```diff
@@ -0,0 +1,387 @@
+
+The kubeadm deb package ships with configuration for how the kubelet should
+be run. Note that the `kubeadm` CLI command will never touch this drop-in file.
+This drop-in file belongs to the kubeadm deb/rpm package.
+
+This is what it looks like:
+
+
+```
+[Service]
+Environment="KUBELET_KUBECONFIG_ARGS=--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf --kubeconfig=/etc/kubernetes/kubelet.conf"
+Environment="KUBELET_SYSTEM_PODS_ARGS=--pod-manifest-path=/etc/kubernetes/manifests --allow-privileged=true"
+Environment="KUBELET_NETWORK_ARGS=--network-plugin=cni --cni-conf-dir=/etc/cni/net.d --cni-bin-dir=/opt/cni/bin"
+Environment="KUBELET_DNS_ARGS=--cluster-dns=10.96.0.10 --cluster-domain=cluster.local"
+Environment="KUBELET_AUTHZ_ARGS=--authorization-mode=Webhook --client-ca-file=/etc/kubernetes/pki/ca.crt"
+Environment="KUBELET_CADVISOR_ARGS=--cadvisor-port=0"
+Environment="KUBELET_CERTIFICATE_ARGS=--rotate-certificates=true --cert-dir=/var/lib/kubelet/pki"
+ExecStart=
+ExecStart=/usr/bin/kubelet $KUBELET_KUBECONFIG_ARGS $KUBELET_SYSTEM_PODS_ARGS $KUBELET_NETWORK_ARGS $KUBELET_DNS_ARGS $KUBELET_AUTHZ_ARGS $KUBELET_CADVISOR_ARGS $KUBELET_CERTIFICATE_ARGS $KUBELET_EXTRA_ARGS
+```
+
+A breakdown of what/why:
+
+* `--bootstrap-kubeconfig=/etc/kubernetes/bootstrap-kubelet.conf` path to a kubeconfig 
+   file that will be used to get client certificate for kubelet during node join. 
+   On success, a kubeconfig file r is written to the path specified by --kubeconfig. 
+* `--kubeconfig=/etc/kubernetes/kubelet.conf` points to the kubeconfig file that
+   tells the kubelet where the API server is. This file also has the kubelet's
+   credentials.
+* `--pod-manifest-path=/etc/kubernetes/manifests` specifies from where to read
+   Static Pod manifests used for spinning up the control plane
+* `--allow-privileged=true` allows this kubelet to run privileged Pods
+* `--network-plugin=cni` uses CNI networking
+* `--cni-conf-dir=/etc/cni/net.d` specifies where to look for the
+   [CNI spec file(s)](https://github.com/containernetworking/cni/blob/master/SPEC.md)
+* `--cni-bin-dir=/opt/cni/bin` specifies where to look for the actual CNI binaries
+* `--cluster-dns=10.96.0.10` use this cluster-internal DNS server for `nameserver`
+   entries in Pods' `/etc/resolv.conf`
+* `--cluster-domain=cluster.local` uses this cluster-internal DNS domain for
+   `search` entries in Pods' `/etc/resolv.conf`
+* `--client-ca-file=/etc/kubernetes/pki/ca.crt` authenticates requests to the Kubelet
+   API using this CA certificate.
+* `--authorization-mode=Webhook` authorizes requests to the Kubelet API by `POST`-ing
+   a `SubjectAccessReview` to the API Server
+* `--cadvisor-port=0` disables cAdvisor from listening to `0.0.0.0:4194` by default.
+   cAdvisor will still be run inside of the kubelet and its API can be accessed at
+   `https://{node-ip}:10250/stats/`. If you want to enable cAdvisor to listen on a
+   wide-open port, run:
```

Comment:
I really like that we take the time to document this, but I think there's a risk of duplicating existing documentation. What if we just link to the kubelet ref docs? https://kubernetes.io/docs/admin/kubelet/

Reply:
I think this is ok; may require an update every quarter which is kinda cheap in the end